### ISSUE TITLE
[clang-tidy] SimplifyDeMorgan of `readability-simplify-boolean-expr` should not trigger within macro

### ISSUE THREAD

**Issue by zero9178** [llvm/llvm-project#57975]
Following code:
```cpp
#include <cstdio>

#define A_ASSERT(...) \
if (!(__VA_ARGS__)) \
    fprintf(stderr, #__VA_ARGS__);

void foo(bool a, bool b)
{
    A_ASSERT(a != true && b == false);
}
```
causes clang-tidy to emit:
```
[<source>:9:5: warning: boolean expression can be simplified by DeMorgan's theorem [readability-simplify-boolean-expr]](javascript:;)
    A_ASSERT(a != true && b == false);
    ^
[<source>:4:5: note: expanded from macro 'A_ASSERT'](javascript:;)
if (!(__VA_ARGS__)) \
    ^~~~~~~~~~~~~~
```
when run with `readability-simplify-boolean-expr` enabled.
Ideally it would not emit that diagnostic as it is impossible to fix, since the inversion that is part of the expression is part of the macro and essential to its function in other expansions. 

CE link:
https://godbolt.org/z/5bc993a1q

**Comment by qqiangwu** [comment]
@njames93 I met this problem too, do you have time to fix it? Or I can fix it.

**Comment by njames93** [comment]
I think it's just a case of ignore macros
